bifacial_radiance/bifacial_radiance.py

```diff
@@ -1130,6 +1130,34 @@ def makeModule(self,name=None,x=1,y=1,bifi=1,orientation='portrait', modulefile=
             json.dump(data,configfile)
         
         print('Module {} successfully created'.format(name))
+
+
+    def makeCustomObject(self,name=None, text=None):
+        '''
+        Function for development and experimenting with extraneous objects in the scene.
+        This function creates a name.rad textfile in the objects folder
+        with whatever text that is passed to it.
+        It is up to the user to pass the correct radiance format.
+        For example, to create a box at coordinates 0,0 (with its bottom surface 
+        on the plane z=0),
+        name = 'box'
+        text='! genbox black PVmodule 0.5 0.5 0.5 | xform -t -0.25 -0.25 0'
+        
+        Parameters
+        ------------
+        name: string input to name the module type
+        text = ''    # text used in the radfile to generate the module
+        '''
+        
+        customradfile = os.path.join('objects','%s.rad'%(name) ) # update in 0.2.3 to shorten radnames
+        # py2 and 3 compatible: binary write, encode text first
+        with open(customradfile, 'wb') as f:
+            f.write(text.encode('ascii'))
+            
+        print("\nCustom Object Name", customradfile)
+        self.customradfile = customradfile
```

Comment:
this can be replaced with self.radfiles.append(customradfile).  Then when self.getfilelist() is called, it will automatically pull the radfile name into the list of files being passed to makeOct.  your appendtoScene function is not necessary then...

Reply:
ok, that would work. And then if you want to get rid of the element you re-run the object.rad file and set it to empty, rerun the makeOct.

the makeCustomObject would still be useful.

Reply:
Sorry, I misunderstood what you were up to - this looks correct.